fesvr/elfloader.cc

```diff
@@ -82,11 +82,50 @@ std::map<std::string, uint64_t> load_elf(const char* fn, memif_t* memif, reg_t*
       } \
     } \
   } while(0)
+  #define LOAD_ELF_NO_MEM(ehdr_t, phdr_t, shdr_t, sym_t, bswap) do { \
```

Comment:
I saw the only difference is _NO_MEM version didn't load anything to memory, just load the symbol table, so add a parameter to LOAD_ELF instead of copy-paste would be better.

```c
    for (unsigned i = 0; i < bswap(eh->e_phnum); i++) {			\
      if(bswap(ph[i].p_type) == PT_LOAD && bswap(ph[i].p_memsz)) {	\
        if (bswap(ph[i].p_filesz)) {					\
          assert(size >= bswap(ph[i].p_offset) + bswap(ph[i].p_filesz)); \
          memif->write(bswap(ph[i].p_paddr), bswap(ph[i].p_filesz), (uint8_t*)buf + bswap(ph[i].p_offset)); \
        } \
        zeros.resize(bswap(ph[i].p_memsz) - bswap(ph[i].p_filesz)); \
        memif->write(bswap(ph[i].p_paddr) + bswap(ph[i].p_filesz), bswap(ph[i].p_memsz) - bswap(ph[i].p_filesz), &zeros[0]); \
      } \
    } \
```

Reply:
Thanks for the reviews！
I’m the mentor of mengls666. We will merge it into LOAD_ELF as what you suggest.